lib/main/widgets/add_pet_button.dart

```diff
@@ -0,0 +1,55 @@
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/widgets.dart';
+import 'package:google_fonts/google_fonts.dart';
+
+class AddPetButton extends StatelessWidget {
+  const AddPetButton({Key? key}) : super(key: key);
```

Comment:
What do you think about adding a callback to this widget? I think you do remember that someone was quite sad when there wasn't one)

```suggestion
class AddPetButton extends StatelessWidget {
  final VoidCallback onTap;

  const AddPetButton({Key? key, this.onTap}) : super(key: key);
```